typescript/cli/src/config/ism.ts

```diff
@@ -87,116 +90,191 @@ export async function createIsmConfig(
+    const relayer = await detectAndConfirmOrPrompt(
+      async () => context.signer?.getAddress(),
+      'For trusted relayer ISM, enter',
+      'relayer address',
+    );
+    return {
+      type: IsmType.TRUSTED_RELAYER,
+      relayer,
+    };
+  },
+  IsmType.TRUSTED_RELAYER,
+);
+
+export const createAggregationConfig = callWithConfigCreationLogsAsync(
+  async (context: CommandContext): Promise<AggregationIsmConfig> => {
+    const isms = parseInt(
+      await input({
+        message: 'Enter the number of ISMs to aggregate (number)',
+      }),
+      10,
+    );
+
+    const threshold = parseInt(
+      await input({
+        message: 'Enter the threshold of ISMs for verification (number)',
+      }),
+      10,
+    );
+
+    const modules: Array<IsmConfig> = [];
+    for (let i = 0; i < isms; i++) {
+      modules.push(await createIsmConfig(context));
+    }
+    return {
+      type: IsmType.AGGREGATION,
+      modules,
+      threshold,
+    };
+  },
+  IsmType.AGGREGATION,
+);
+
+export const createRoutingConfig = callWithConfigCreationLogsAsync(
+  async (context: CommandContext): Promise<IsmConfig> => {
+    const owner = await input({
+      message: 'Enter owner address for routing ISM',
+    });
+    const ownerAddress = owner;
+
+    const chains = await runMultiChainSelectionStep(
+      context.chainMetadata,
+      'Select chains to configure routing ISM for',
+      true,
+    );
+
+    const domainsMap: ChainMap<IsmConfig> = {};
+    for (const chain of chains) {
+      await confirm({
+        message: `You are about to configure routing ISM from source chain ${chain}. Continue?`,
+      });
+      const config = await createIsmConfig(context);
+      domainsMap[chain] = config;
+    }
+    return {
+      type: IsmType.ROUTING,
+      owner: ownerAddress,
+      domains: domainsMap,
+    };
+  },
+  IsmType.ROUTING,
+);
+
+export const createFallbackRoutingConfig = callWithConfigCreationLogsAsync(
+  async (context: CommandContext): Promise<IsmConfig> => {
+    const owner = await input({
+      message: 'Enter owner address for fallback routing ISM',
     });
-    const config = await createIsmConfig(context);
-    domainsMap[chain] = config;
+    const ownerAddress = owner;
+
+    const chains = await runMultiChainSelectionStep(
+      context.chainMetadata,
+      'Select chains to configure fallback routing ISM for',
+      true,
+    );
+
+    const domainsMap: ChainMap<IsmConfig> = {};
+    for (const chain of chains) {
+      await confirm({
+        message: `You are about to configure fallback routing ISM from source chain ${chain}. Continue?`,
+      });
+      const config = await createIsmConfig(context);
+      domainsMap[chain] = config;
+    }
+    return {
+      type: IsmType.FALLBACK_ROUTING,
+      owner: ownerAddress,
+      domains: domainsMap,
+    };
+  },
+  IsmType.FALLBACK_ROUTING,
+);
+
+export async function createIsmConfigWithWarningOrDefault({
```

Comment:
nit: name need not include implementation detail of "warning or default"?

Reply:
yeah this was mostly to get around the already present `createIsmConfig` function